solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRCollector.java

```diff
@@ -128,7 +129,10 @@ public TopDocs topDocs(int start, int howMany) {
 
       return topRerankDocs;
 
-    } catch (final Exception e) {
+    } catch (final FeatureException e) {
+      throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, e);
+    }
+    catch (final Exception e) {
```

Comment:
A FeatureException  is a regular exception, this catch should be able to handle it for you without the extra case